### PR TITLE
Allow installing Monolog 3, and run test suite with it

### PR DESCRIPTION
**Pull request by Seldaek**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
             fail-fast: false
             matrix:
                 php: [ '7.2', '7.3', '7.4', '8.0', '8.1' ]
-                monolog: [ '1.*', '2.*' ]
+                monolog: [ '1.*', '2.*', '3.*@dev' ]
```
Comment (derrabus) [inline .github/workflows/ci.yaml:15]: This matrix spawns jobs with PHP 7.2 and Monolog 3 which is pointless.

Instead, let's add an entry to the `include` section:

```YAML
-   php: '8.1'
    monolog: '3.x@dev'
```

Comment (Seldaek) [reply .github/workflows/ci.yaml]: Yeah.. this was a rush job and it shows 😅 Thanks

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -266,7 +270,7 @@ public function testWithConsoleHandler()
         $this->assertSame([
             OutputInterface::VERBOSITY_NORMAL => Logger::NOTICE,
             OutputInterface::VERBOSITY_VERBOSE => Logger::INFO,
-            OutputInterface::VERBOSITY_VERY_VERBOSE => 150,
+            OutputInterface::VERBOSITY_VERY_VERBOSE => 200,
```
Comment (stof) [inline Tests/DependencyInjection/ConfigurationTest.php:273]: shouldn't this use a constant ?

Comment (Seldaek) [reply Tests/DependencyInjection/ConfigurationTest.php:273]: Yes it should but I assume it was testing explicitly with a string value as technically people can use those in the config files. Anyway I just fixed it to have a valid level as 150 does not exist.

**Comment by Seldaek** [comment]
Note that the 1.*/2.* failures are not related to this PR I believe - seems to be some deprecation warnings from symfony/monolog-bridge 5.4

**Review by derrabus — APPROVED** [review]

**Pull request merged**